values.yaml

```diff
@@ -450,6 +450,8 @@ core:
   affinity: {}
   ## Additional deployment annotations
   podAnnotations: {}
+  ## Additional environment variables
+  containerEnv: []
```

Comment:
should this be `containerEnv: []`?

Reply:
I think it should be a list based on the API [docs](https://kubernetes.io/docs/reference/generated/kubernetes-api/v1.21/#container-v1-core)

> <table>
> <thead><tr><th>Field</th><th>Description</th></tr></thead>
> <tbody>
> <tr><td><code>env</code><br><i><a href="https://kubernetes.io/docs/reference/generated/kubernetes-api/v1.21/#envvar-v1-core">EnvVar</a> array</i><br><b>patch strategy</b>: <i>merge</i><br><b>patch merge key</b>: <i>name</i></td><td>List of environment variables to set in the container. Cannot be updated.</td></tr>
> </tbody>
> </table>

Reply:
by the way it is injected, wonder how array or map would fit into the listentry expected having
- name
- value

lemme suggest:
```
{{- if .Values.core.containerEnv }}
- name: CONFIG_OVERWRITE_JSON
- value: {{ .Values.core.containerEnv  }}
{{- end }}
```

thereby .Values.core.containerEnv is expected string, otherwise you would probably need yaml2json.

Reply:
I think this is a good idea. In this case I'd suggest changing the name to something like `.Values.core.configOverwriteJson`. Your change limits this feature to only modifying the CONFIG_OVERWRITE_JSON environment variable instead of allowing arbitrary environment variables to be injected. This might be a good thing.

I don't know of any other environment variables someone would want to set. If there are multiple environment variables that make sense to set, then my way makes more sense. If CONFIG_OVERWRITE_JSON is the only one, then your change makes more sense. I'm happy to use your suggestion. @janesser @zyyw what do you think?